### ISSUE TITLE
Arch Linux

### ISSUE THREAD

**Issue by ruby-rust-perl** [jupyterlab/jupyterlab_app#251]
Is it possible to include a package for Arch Linux for example in AUR or just in `snap`?

**Comment by krassowski** [comment]
`snap` or `flatpak` sound tempting as cross-platform installers that could reduce the maintenance burden, but I understand that these may still provide not-fully native experience on some distributions. How is the experience on Arch with these two?

**Comment by ruby-rust-perl** [comment]
Yeah. I agree with you. They don't provide a native feel. I have never used `flatpak`. I have only used `snap`. I used it to install `VSCode`, `Brave`, `powershell` (`pwsh`), `IntelliJ` etc. I haven't faced issues with any of them till now. They work perfectly except for few commands in `pwsh` (which can't be attributed to `snap` though). The reason I use `snap` even if it ain't native because it is easier to install packages. All of the packages I installed using `snap` didn't require me running commands to install dependencies. They were all one-liners `sudo snap install something [--classic]`. They were all in this format. So, it reduces the user's effort in installing the dependencies manually. Sometimes, the versions may conflict when installed manually and there are many other complications. In short, `snap` is really good and my experience with `snap` is smooth so far

**Comment by mbektas** [comment]
@ruby-rust-perl we are using [electron-builder](https://www.electron.build/) to create our installers and it has some limitations on post install script execution. we run a script after App installation to setup the Python environment and `electron-builder` supports it for Debian and RPM installers as far as I could figure out. That's why our Linux installers are limited to these platforms. More information on [electron-builder Linux configuration](https://www.electron.build/configuration/linux.html)

Once we add support for connecting to existing Python environments, we can release JupyterLab App for more Linux platforms since we can optionally release the App without a Python environment.

**Comment by manojanasuri16** [comment]
I really like using the app natively on my machine, so I've compiled the app for arch Linux from the Debian package
I know it was best to distribute a snap or flatpak as a cross-platform installer.
I'm a newbie trying out different things.
The app was working fine for me.
Link for the app: https://github.com/manojanasuri16/jupyterlab_app/releases/download/v3.1.12-1/jupyterlab-app-3.1.12-1-x86_64.zst

**Comment by mbektas** [comment]
@manojanasuri16 thanks for your interest in helping out with Arch Linux version. Please create a PR with your code changes to create the installer for the platform and please do not paste links to installer binaries.

**Comment by giswqs** [comment]
I was able to compile and install the app on the Arch-based Manjaro Linux. Excited to see the app runing!

```
yay -S debtap
debtap JupyterLab-Setup-Debian.deb
sudo pacman -U jupyterlab-app-3.1.13-1-x86_64.pkg.tar.zst
```

![image](https://user-images.githubusercontent.com/5016453/135189738-983a0019-675d-4b69-8596-0556ebabdf76.png)

![image](https://user-images.githubusercontent.com/5016453/135189941-a37ed06a-0335-414d-910e-476ca802be00.png)

**Comment by matheusfillipe** [comment]
I installed it like @giswqs described on archlinux and when I try to launch it it just stays on the loading screen forever.
![image](https://user-images.githubusercontent.com/24435787/135951171-55c855c9-04dd-41a9-bd5d-2a28956f1499.png)


I tried launching like this from the terminal: `/opt/JupyterLab/jupyterlab_app -v` willing to get some info but nothing was displayed. Is there another way I could install this on arch? (I am going to give a try on building it soon, maybe someone could come with a AUR package)
